.circleci/config.yml

```diff
@@ -39,7 +39,7 @@ jobs:
         type: string
 
     docker:
-      - image: cimg/python:3.9
+      - image: cimg/python:3.12
```

Comment:
As @randomir points out in https://github.com/dwavesystems/dwave-cloud-client/pull/671#issue-2564499068, we could do standard Python images. I was contemplating switching over next time I touched these in `dwave-optimization` etc.

Reply:
Guess I might as well do that here as well.

Reply:
Got a `FileNotFoundError: [Errno 2] No such file or directory: 'docker'` error when switching to the `python:3.12` image, so kept the `cimg/python` one for now. Will look into, and update, later.